src/main/java/com/vmware/vipclient/i18n/base/instances/LocaleMessage.java

```diff
@@ -33,8 +43,15 @@ public LocaleMessage() {
      * @return map the key is the supported language tag, value is all region data mapped by
      *         language tag
      */
-    public Map<String, Map<String, String>> getRegionList(List<String> supportedLanguageList) {
-        return new LocaleService().getTerritoriesFromCLDR(supportedLanguageList);
+    public Map<String, Map<String, String>> getRegionList(List<String> localeList) {
+        Map<String, Map<String, String>> respMap = new HashMap<String, Map<String, String>>();
+        for (String locale : localeList) {
+            if(locale != null && !locale.isEmpty()) {
+                Map<String, String> regionMap = getRegionsFromCLDR(locale);
```

Comment:
As we discussed offline, add a new class LocaleDTO in case parameter increase, but won’t put locale into it.
Because 'locale' is a frequently changed parameter, put it into DTO will need new service instance multiple times when its value changed, and new instance is time consuming.